cmd/createorupdate/update.go

```diff
@@ -80,70 +80,68 @@ func updatePlusOne(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.P
 	// store a list of all the VM instances now, so that if we end up creating
 	// new ones (in the crash recovery case, we might not), we can detect which
 	// they are
-	vms, err := listVMs(ctx, cs, vmc, role)
+	oldVMs, err := listVMs(ctx, cs, vmc, role)
 	if err != nil {
 		return err
 	}
 
+	// TODO: Filter out VMs that do not need to get upgraded. Should speed
+	// up retrying failed upgrades.
 	vmsBefore := map[string]struct{}{}
-	for _, vm := range vms {
+	for _, vm := range oldVMs {
 		vmsBefore[*vm.InstanceID] = struct{}{}
 	}
+	vmsUpgraded := map[string]struct{}{}
```

Comment:
why is this needed, out of interest? any reason you can't add new vms into vmsBefore as you iterate?

Reply:
I think you are right. I used to iterate over vmsBefore but not anymore so I can reuse it.